tests/log_collector_test.go

```diff
@@ -0,0 +1,272 @@
+			} else {
+				t.Logf("Note: Could not verify file uploads (this is OK in mock mode): %v", execErr)
+			}
+		}
+
+		return ctx
+	})
+
+	feature.Teardown(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		return helpers.TeardownFaultRemediation(ctx, t, c)
+	})
+
+	testEnv.Test(t, feature.Feature())
+}
+
+// TestLogCollectorFailure tests that log collector job fails correctly in mock mode
+func TestLogCollectorFailure(t *testing.T) {
+	t.Skip("Skipping failure test - requires Helm chart redeployment with mockExitCode=1")
+
+	feature := features.New("Log Collector - Failure Scenario").
+		WithLabel("suite", "log-collector").
+		WithLabel("component", "mock-mode").
+		WithLabel("scenario", "failure")
+
+	var testCtx *helpers.RemediationTestContext
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		var newCtx context.Context
+		newCtx, testCtx = helpers.SetupFaultRemediationTest(ctx, t, c, "")
+		return newCtx
+	})
+
+	feature.Assess("Trigger fault and verify log collector job fails", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		client, err := c.NewClient()
+		require.NoError(t, err)
+
+		// Trigger fault remediation flow
+		helpers.TriggerFullRemediationFlow(ctx, t, client, testCtx.NodeName, 2)
+
+		// Wait for log collector job to be created
+		t.Logf("Waiting for log collector job to be created for node %s", testCtx.NodeName)
+		var logCollectorJob *batchv1.Job
+		require.Eventually(t, func() bool {
+			jobs := &batchv1.JobList{}
+			err := client.Resources(helpers.NVSentinelNamespace).List(ctx, jobs)
+			if err != nil {
+				return false
+			}
+
+			for _, job := range jobs.Items {
+				if strings.Contains(job.GetName(), "log-collector") {
+					logCollectorJob = &job
+					return true
+				}
+			}
+			return false
+		}, helpers.EventuallyWaitTimeout, helpers.WaitInterval, "log collector job should be created")
+
+		require.NotNil(t, logCollectorJob, "log collector job should exist")
+		t.Logf("Log collector job created: %s", logCollectorJob.GetName())
+
+		// Wait for job to fail (mock mode with exitCode=1)
+		t.Logf("Waiting for log collector job to fail")
+		require.Eventually(t, func() bool {
+			job := &batchv1.Job{}
+			err := client.Resources(helpers.NVSentinelNamespace).Get(ctx, logCollectorJob.GetName(), helpers.NVSentinelNamespace, job)
+			if err != nil {
+				return false
+			}
+			return job.Status.Failed > 0
+		}, helpers.EventuallyWaitTimeout, helpers.WaitInterval, "log collector job should fail")
+
+		t.Logf(" Log collector job failed as expected")
+
+		return ctx
+	})
+
+	feature.Teardown(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		return helpers.TeardownFaultRemediation(ctx, t, c)
+	})
+
+	testEnv.Test(t, feature.Feature())
+}
+
+// TestLogCollectorEndToEnd tests the complete end-to-end flow with log collector
+func TestLogCollectorEndToEnd(t *testing.T) {
```

Comment:
This seems to overlap with the other two test cases, can you check? Can we have a small number of clean targeted tests?

Reply:
> So which approach is better?

Single test for happy path, one dedicated test case for failure